pkg/docker/config/config.go

```diff
@@ -136,6 +136,11 @@ func getAuthFilePaths(sys *types.SystemContext) []authPath {
 		// Logging the error as a warning instead and moving on to pulling the image
 		logrus.Warnf("%v: Trying to pull image in the event that it is a public image.", err)
 	}
+	if dockerConfig := os.Getenv("DOCKER_CONFIG"); dockerConfig != "" {
+		paths = append(paths,
+			authPath{path: filepath.Join(dockerConfig, "config.json"), legacyFormat: false},
+		)
+	}
```

Comment:
Looking at https://github.com/projectatomic/docker/blob/master/cliconfig/config.go as well as more recent versions, the `DOCKER_CONFIG` path _replaces_ the $HOME`/.docker`, it does not add one more file to check. So this code should do the same thing.